rfcs/text/0032-structured-benchmark-programs.md

```diff
@@ -0,0 +1,157 @@
+
+However, a large number of existing quantum compiler toolchains still do not support some or even all of these features.
+Proposing a set of such programs as benchmarks can not only help users evaluate the capabilities of different toolchains, but also serve as a motivation for compiler developers to implement these features.
+
+Furthermore, it can also increase the visibility of Jeff as a quantum program format that can express these advanced features, encouraging more users to adopt it for their quantum programming needs.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+A new directory, `benchmarks/structured` is introduced in the Jeff repository.
+It provides a set of individual Jeff files that each represent a single structured benchmark program.
+
+In cases where the programs are adaptive in their size, it uses dynamic input parameters to keep the program implementation agnostic of the specifically desired size.
+This may include examples such as arbitrarily sized GHZ state preparation, where the number of qubits is specified as an input parameter at runtime.
+Similarly, programs that require classical input data such as rotation angles use input parameters to allow users to specify the data at runtime.
+An example of this might be a VQA ansatz that is repeated several times using `for` loops.
+
+In contrast, some programs are provided with fixed sizes and parameters, either because they cannot easily be generalized or because they are intended as simpler test scenarios for basic compiler functionality.
+Examples of this include simple Quantum Teleportation programs, or predefined instances of Grover's search algorithm.
+
+The modular directory structure allows future adaptations and extensions of the benchmark suite, either by adding other structured programs or by also including entirely new categories of benchmarks.
+
+While this feature may not directly impact the way many users interact with Jeff, it provides a valuable resource for evaluating and improving quantum compilers and test their compatibility with Jeff and advanced quantum programming features.
+
+# Implementation-level explanation
+[implementation-level-explanation]: #implementation-level-explanation
+
+The following list outlines the proposed structured benchmark programs to be included in the `benchmarks/structured` directory, along with brief description, grouped by the way they utilize structured control flow:
+
+*Note: this list was created from previous discussions among community members and is open for further suggestions and modifications.*
+
+## Static Loops without Dynamic Qubit Indexing
+
+*This class of benchmark programs utilizes loops (such as `for` and `while`) that **do not** depend on dynamic execution results. Inside the loop bodies, all operations are still statically indexed.*
```

Comment:
Is the term "dynamic execution results" established enough in quantum?
I would have probably went into the direction of "do not use mid-circuit measurements and conditional quantum operations", or similar.

Reply:
I was confused by this as well. I'm leaving a related comment in the "Conditionals" section.

Reply:
Dynamic program values instead of dynamic execution results might be a bit clearer imo, or something runtime-computed program values.